modules/dr-recover-lost-control-plane-hosts.adoc

```diff
@@ -0,0 +1,301 @@
+----
+2019-05-20 15:33:17.648445 E | rafthttp: request cluster ID mismatch (got 9f5f9f05e4d43b7f want 807ae3bffc8d69ca)
+----
+====
+
+. Create new master hosts.
++
+If your cluster has its Machine API enabled and functional, then when the
+OpenShift `machine-api` Operator is restored, it will create the new masters. If you do
+not have the `machine-api` Operator enabled, you must create new masters
+using the same methods that were used to originally create them.
++
+In a terminal that has access to the cluster, run the following command to verify that your masters are ready:
++
+----
+$ oc get nodes -l node-role.kubernetes.io/master
+NAME                                         STATUS   ROLES    AGE   VERSION
+ip-10-0-143-125.us-east-2.compute.internal   Ready    master   50m   v1.13.4+db7b699c3
+ip-10-0-156-255.us-east-2.compute.internal   Ready    master   92s   v1.13.4+db7b699c3
+ip-10-0-162-178.us-east-2.compute.internal   Ready    master   70s   v1.13.4+db7b699c3
+----
+
+. Correct the DNS entries.
+
+.. Obtain the private IP address of the new instance.
++
+In a terminal that has access to the cluster, run the following command:
++
+----
+$ oc get node ip-10-0-143-125.us-east-2.compute.internal -o  jsonpath='{.status.addresses[?(@.type=="InternalIP")].address}{"\n"}'
+10.0.143.125
+----
+
+.. From the AWS console, edit the etcd-0 Route 53 record in the private DNS zone and set the value to the new private IP address. See link:https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-editing.html[Editing Records] in the AWS documentation for instructions.
+
+.. Repeat these steps to update the etcd-1 and etcd-2 entries as appropriate to the current addresses.
+
+. Update load balancer entries.
++
+If you are using a cluster-managed load balancer, the entries will automatically be updated for you. If you are not, be sure to update your load balancer with the current addresses of your master hosts.
++
+If your load balancing is managed by AWS, see link:https://docs.aws.amazon.com/elasticloadbalancing/latest/application/target-group-register-targets.html#register-ip-addresses[Register or Deregister Targets by IP Address] in the AWS documentation for instructions on updating load balancer entries.
+
+. Grow etcd to full membership.
+
+.. Set up a temporary etcd certificate signer service on your master where you have restored etcd.
+
+... Access the original master, and log in to your cluster as a `cluster-admin` user using the following command.
++
+----
+[core@ip-10-0-143-125 ~]$ oc login https://localhost:6443
+Authentication required for https://localhost:6443 (openshift)
+Username: kubeadmin
+Password:
+Login successful.
+----
+
+... Obtain a local copy of the pull secret used to install the cluster.
++
+----
+[core@ip-10-0-143-125 ~]$ oc get secret pull-secret -n openshift-config -o jsonpath={.data.'\.dockerconfigjson'} | base64 -d > config.json
+----
```

Comment:
As noted recently in google doc  and in other examples the pull secret is already on disk so we can skip this part and ref the file on disk.

--registry-config='/var/lib/kubelet/config.json'